src/simulator_core/infrastructure/plotting.py

```diff
@@ -0,0 +1,112 @@
+        """Function to prepare geopandas dataframe."""
+        geo_df = gpd.GeoDataFrame(
+            columns=['asset_id', 'asset_name', 'geometry', 'mass_flow', 'pressure_in',
+                     'pressure_out', 'temperature_in', 'temperature_out'], crs="EPSG:4326")
+
+        assets = self.get_assets_by_asset_type(esdl.Pipe)
+
+        for asset in assets:
+            if isinstance(asset.geometry, esdl.esdl.Line):
+                coor_pipe = []
+                for point in asset.geometry.point:
+                    coor_pipe.append([point.lon, point.lat])
+
+                last_data_index = len(result) - 1
+                data = {'asset_id': asset.id,
+                        'asset_name': asset.name,
+                        'geometry': LineString(coor_pipe),
+                        'mass_flow': result[(asset.id, 'mass_flow')][last_data_index],
+                        'pressure_in': result[(asset.id, 'pressure_supply')][last_data_index],
+                        'pressure_out': result[(asset.id, 'pressure_return')][last_data_index],
+                        'temperature_in': result[(asset.id, 'temperature_supply')][last_data_index],
+                        'temperature_out': result[(asset.id, 'temperature_return')][
+                            last_data_index],
+                        }
+
+                geo_df.loc[len(geo_df)] = data
+
+        return geo_df
+
+    @staticmethod
+    def plot_line_mapbox(geo_df: gpd.GeoDataFrame) -> None:
+        """Function to plot using plotly express line mapbox."""
+        lats: NDArray = np.array(None)
+        lons: NDArray = np.array(None)
+        names: NDArray = np.array(None)
+        for _, row in geo_df.iterrows():
+            if isinstance(row['geometry'], LineString):
+                linestrings = [row['geometry']]
+            else:
+                continue
+            for linestring in linestrings:
+                x, y = linestring.xy
+                lats = np.append(lats, y)
+                lons = np.append(lons, x)
+
+                text = row['asset_name'] + \
+                    ", mass_flow: " + "{:.2f}".format(row['mass_flow']) + " kg/s" + \
+                    ", Pin: " + "{:.2f}".format(row['pressure_in'] / 1e5) + " bar" + \
+                    ", Pout: " + "{:.2f}".format(row['pressure_out'] / 1e5) + " bar" + \
+                    ", Tin: " + "{:.2f}".format(row['temperature_in'] - 273) + " C" + \
+                    ", Tout: " + "{:.2f}".format(row['temperature_out'] - 273) + " C"
+
+                names = np.append(names, [text] * len(y))
+                lats = np.append(lats, np.array(None))
+                lons = np.append(lons, np.array(None))
+                names = np.append(names, np.array(None))
+
+        fig = px.line_mapbox(lat=lats, lon=lons,
+                             hover_name=names, zoom=14, mapbox_style="carto-positron")
+        fig.show()
+
+    def get_assets_by_asset_type(self, asset_type: esdl) -> list[esdl.Asset]:
```

Comment:
You have an error for the type indication here.

Reply:
what do you mean? all checks are passed, including type check